metrics.go

```diff
@@ -1293,7 +1293,13 @@ func (m *RPCMetrics) Merge(other *RPCMetrics) {
 		}
 		existing, ok := m.LastDay[k]
 		if !ok {
-			m.LastDay[k] = v
+			// Deep copy to avoid sharing slice references
+			vCopy := v
+			if len(v.Segments) > 0 {
+				vCopy.Segments = make([]RPCStats, len(v.Segments))
+				copy(vCopy.Segments, v.Segments)
```

Comment:
Let's just make it simpler....
```suggestion
				vCopy.Segments = append([]RPCStats{}, v.Segments...)
```